File: src/main/java/kr/hahaha98757/zombiesaddon/commands/CommandMap.java
package kr.hahaha98757.zombiesaddon.commands;

import kr.hahaha98757.zombiesaddon.utils.Utils;
import net.minecraft.client.Minecraft;
import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;

public abstract class CommandMap extends CommandBase {
    public abstract String getMap();

    @Override
    public int getRequiredPermissionLevel() {
        return 0;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "/" + getCommandName();
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if (sender != Minecraft.getMinecraft().thePlayer) return;
        Utils.sendChat("/play arcade_zombies_" + getMap());
    }
}

class CommandDE extends CommandMap {
    @Override
    public String getCommandName() {
        return "de";
    }

    @Override
    public String getMap() {
        return "dead_end";
    }
}

class CommandBB extends CommandMap {
    @Override
    public String getCommandName() {
        return "bb";
    }

    @Override
    public String getMap() {
        return "bad_blood";
    }
}

class CommandAA extends CommandMap {
    @Override
    public String getCommandName() {
        return "aa";
    }

    @Override
    public String getMap() {
        return "alien_arcadium";
    }
}

class CommandPR extends CommandMap {
    @Override
    public String getCommandName() {
        return "pr";
    }

    @Override
    public String getMap() {
        return "prison";
    }
}

File: src/main/kotlin/kr/hahaha98757/zombiesaddon/commands/CommandMap.kt
package kr.hahaha98757.zombiesaddon.commands

import kr.hahaha98757.zombiesaddon.utils.sendChat
import net.minecraft.command.ICommandSender

abstract class CommandMap: CustomCommandBase() {
    abstract fun getMap(): String
    final override fun getCommandUsage(sender: ICommandSender?) = "/$commandName"
    final override fun runCommand(sender: ICommandSender, args: Array<String>) = sendChat("/play arcade_zombies_${getMap()}")
}

object CommandDE: CommandMap() {
    override fun getCommandName() = "de"
    override fun getMap() = "dead_end"
}

object CommandBB: CommandMap() {
    override fun getCommandName() = "bb"
    override fun getMap() = "bad_blood"
}

object CommandAA: CommandMap() {
    override fun getCommandName() = "aa"
    override fun getMap() = "alien_arcadium"
}

object CommandPR: CommandMap() {
    override fun getCommandName() = "pr"
    override fun getMap() = "prison"
}